deepchem/splits/splitters.py

```diff
@@ -1408,28 +1456,25 @@ def split(
       Each indices is a list of integers.
     """
     np.testing.assert_almost_equal(frac_train + frac_valid + frac_test, 1.)
-    scaffold_sets = self.generate_scaffolds(dataset)
-
-    train_cutoff = frac_train * len(dataset)
-    valid_cutoff = (frac_train + frac_valid) * len(dataset)
-    train_inds: List[int] = []
-    valid_inds: List[int] = []
-    test_inds: List[int] = []
-
-    logger.info("About to sort in scaffold sets")
-    for scaffold_set in scaffold_sets:
-      if len(train_inds) + len(scaffold_set) > train_cutoff:
-        if len(train_inds) + len(valid_inds) + len(scaffold_set) > valid_cutoff:
-          test_inds += scaffold_set
-        else:
-          valid_inds += scaffold_set
-      else:
-        train_inds += scaffold_set
+
+    if seed is not None:
+      np.random.seed(seed)
+
+    scaffold_sets = self.generate_scaffolds(dataset, log_every_n)
+
+    expected_train_size = int(round(frac_train * len(dataset)))
+    expected_valid_size = int(round(frac_valid * len(dataset)))
+    expected_test_size = len(
+        dataset) - expected_train_size - expected_valid_size
+    train_inds, valid_inds, test_inds = _select_clusters(
+        scaffold_sets, expected_train_size, expected_valid_size,
+        expected_test_size)
+
     return train_inds, valid_inds, test_inds
 
   def generate_scaffolds(self,
                          dataset: Dataset,
-                         log_every_n: int = 1000) -> List[List[int]]:
+                         log_every_n: Optional[int] = 1000) -> List[List[int]]:
```

Comment:
Why is this made optional? Does changing the type break backwards compatibility?

Reply:
Look at the [call of this function](https://github.com/deepchem/deepchem/blob/master/deepchem/splits/splitters.py#L1411). There was a minor bug, so `log_every_n` wasn't used. I fixed that so it needed to be made optional.

Reply:
I think if a default is set it doesn't need to be optional. Making it `Optional` allows for `log_every_n` to be `None`. That isn't something we support at present as a convention. If we make this change, we'd have to make it across the codebase to retain the calling convention